### ISSUE TITLE
Remote computers not seeing each other, initially

### ISSUE THREAD

**Issue by jon-ault** [FoldingAtHome/fah-web-client-bastet#215]
I have two computers running 8.4.9 (one Windows, one Mac) and one Windows computer running 8.3.18. None of the web clients can see any of the other computers, each one can only show its own status with the other two grayed out.
![Screenshot 2024-11-21 at 1 21 08 PM](https://github.com/user-attachments/assets/294e0244-730b-45ae-afa6-40ef19d44e62)

**Comment by jon-ault** [comment]
Yes, logged in on all machines.

**Comment by muziqaz** [comment]
Does Mac have some sort of software firewall? Are you using firewall on Windows machines?

**Comment by muziqaz** [comment]
> Yes, logged in on all machines. Also, when logged out the other machines simply don't appear at all; I have to be logged in to see the greyed out sections for those computers.

FAH is account based. You have to be logged in to see remote machines.

**Comment by jon-ault** [comment]
Yes, the Mac has a built-in firewall. It wasn't causing any issues with 8.3, and turning it off does not fix this issue in 8.4.

As for being logged in, the screenshot above was taken while logged in on all computers.

**Comment by jon-ault** [comment]
OK this is interesting. If I leave the browser running long enough untouched it eventually connects. I'm not sure how long it takes, something on the order of 15-20 minutes. I'll try to time it.

**Comment by jon-ault** [comment]
It took about 4 minutes for the other machines to appear, so I was way off in my initial estimate.
![Screenshot 2024-11-21 at 2 32 09 PM](https://github.com/user-attachments/assets/7c2ac38e-b71a-4117-bbe5-7adf0c453ebe)

**Comment by muziqaz** [comment]
yeah, it seems Firefox is not the favorite browser for FAH :)

That's probably related to this:
https://github.com/FoldingAtHome/fah-client-bastet/issues/287 
above moved to the following issue:
https://github.com/FoldingAtHome/fah-web-client-bastet/issues/214

**Comment by jcoffland** [comment]
These are different issues.  One is about the local client taking a long time to show.  This issue is about seeing remote clients.

**Comment by jcoffland** [comment]
Was this issue also caused by not being on the correct Web Control page?

**Comment by kbernhagen** [comment]
I was using the correct v8-4 web page.
Brave, Safari, mobile Safari are all working normally right now.

**Comment by kbernhagen** [comment]
I considered it a similar issue because you mentioned web control would abort connections early to avoid browser connection fall off in another issue.

**Comment by muziqaz** [comment]
I think this is just initial loading. After that, I never had issues in my case

**Comment by jcoffland** [comment]
> I considered it a similar issue because you mentioned web control would abort connections early to avoid browser connection fall off in another issue.

This is only with the local client connection.  Web Control connects directly to the local client when possible.  The remote connections are all via fah-node.  There's only a delay if you have problems connecting to fah-node which usually only happens when your Internet connection goes down.  fah-node itself is on Amazon AWS and has a very high up time.

**Comment by kbernhagen** [comment]
Then what was causing
`WebSocket is closed before the connection is established`

**Comment by jcoffland** [comment]
That's the Websocket to the local client.

**Comment by kbernhagen** [comment]
No. It was a node connection.
`WebSocket connection to 'wss://node1.foldingathome.org/ws/account' failed: WebSocket is closed before the connection is established.`

**Comment by kbernhagen** [comment]
The local client connected immediately. Remotes did not.
I figured you modified the deployed Web Control or fah-node when it started working again..

**Comment by jcoffland** [comment]
I didn't modify anything.

**Comment by bollix47** [comment]
The problem is back today after working fine overnight and this morning .... i.e. can see local machine but not the remote ones

**Comment by kbernhagen** [comment]
Web control v8-3 seems to have the same problem. I don’t have an older client running right now to test it.

**Comment by muziqaz** [comment]
Firefox can’t establish a connection to the server at wss://node1.foldingathome.org/ws/account. 
The connection to wss://node1.foldingathome.org/ws/account was interrupted while the page was loading.

**Comment by bollix47** [comment]
my remotes have now shown up .... a half hour after I tried to load the page

**Comment by kbernhagen** [comment]
This was happening again today, and started working right now.
Still takes a few seconds for remotes to connect.
Something is not reliable.
Sending a reset command via `lufah` had no effect.

macOS 15.1.1, Safari, Brave

**Comment by jcoffland** [comment]
I think this is working now.
